src/printer/components/projects.js

```diff
@@ -23,152 +22,123 @@ const sortByType = (a, b) => {
+  if (metadata.current_path.length > 0) {
+    let view = metadata.files[metadata.current_path[0]];
+    for (let i = 1; i < metadata.current_path.length; i++) {
+      let path = metadata.current_path[i];
+      view = view.find((elm) => elm.name == path).children;
+    }
+
+    projects.appendChild(createUp());
+    for (let node of view.sort(sortByType)) {
+      if (node.type == "folder") {
+        projects.appendChild(createFolder(node.display));
+      } else {
+        projects.appendChild(createFile(node));
+      }
+    }
+  } else {
+    for (let name in metadata.files) {
+      projects.appendChild(createFolder(name));
+    }
   }
 }
-function showFile(elm, node) {
+
+function createElement(templateName, name, cb) {
+  const templateFolder = document.getElementById(templateName).content;
+  const elm = document.importNode(templateFolder, true);
   elm.querySelector(".node").addEventListener("click", (e) => {
-    metadata.current_view = node;
-    navigate("#preview");
+    cb(e);
     e.preventDefault();
   });
-  elm.querySelector("p").appendChild(document.createTextNode(node.display));
-  const nodeDetails = elm.querySelector(".node-details");
-  for (let property of ["printTime", "material", "layerHeight"]) {
-    let element = elm.getElementById(property);
-    if (node[property] != undefined) {
-      let span = document.createElement("span");
-      span.innerHTML = node[property];
-      element.querySelector("p").appendChild(span);
-    } else {
-      nodeDetails.removeChild(element);
-    }
-  }
+  elm.querySelector("p").appendChild(document.createTextNode(name));
+  return elm;
 }
 
-function showFolder(elm, node) {
-  elm.querySelector(".node").addEventListener("click", (e) => {
-    metadata.current_view = node;
+function createFolder(name) {
+  return createElement("node-folder", name, () => {
+    metadata.current_path.push(name);
     show();
-    e.preventDefault();
   });
-  elm.querySelector("p").appendChild(document.createTextNode(node.display));
 }
 
-function showUp(elm, node) {
-  elm.querySelector(".node").addEventListener("click", (e) => {
-    metadata.current_view = node;
+function createUp() {
+  return createElement("node-up", "Main", () => {
+    metadata.current_path.pop();
     show();
-    e.preventDefault();
   });
-  elm.querySelector("p").appendChild(document.createTextNode(node.display));
 }
 
-export function show() {
-  const view = metadata.current_view || metadata.container;
-  const projects = document.querySelector(".projects");
-
-  while (projects.firstChild) {
-    projects.removeChild(projects.firstChild);
+const selectError = (status, data) => {
+  if (!status.ok) {
+    console.error(`Cant get printer API! Error ${status.code}`);
+    console.error(data);
   }
+};
 
-  if (view.type == "folder") {
-    const templateFolder = document.getElementById("node-folder").content;
-    const templateProject = document.getElementById("node-project").content;
+const onClickFile = (node) => {
```

Comment:
I think that here is being used wrong endpoint. You click on file in list and you expect to just see its details, but by this click you start printing.

Reply:
The endpoint is in node.refs.resource, this should be the path to the file, it's sending a POST request with the body {command: "select" } as in https://docs.octoprint.org/en/master/api/files.html#issue-a-file-command .

Reply:
When the project is open in the printer, it should be available in all the other interfaces. So, it's for select the project in the printer.